### DISCUSSION TITLE
Loss for sparse node classification

### DISCUSSION THREAD

**Discussion by ilyalasy** [pyg-team/pytorch_geometric#3691] [Q&A] [answered]
Hey, is there any research on node classification with imbalanced classes?
In particular I have the idea of changing simple cross entropy loss to some function that is able to score close predictions.
_E.g._ node _n_ has class **A** and it has 3 neighbors that have class **B**, if model predicts that one of these neighbors has class **A**, such loss function should not penalize this prediction as hard as cross entropy.
Is there any similar loss/metric existing?

**Comment by rusty1s** [comment, marked as answer]
In general, people are using weighted cross entropy on imbalanced classes (based on the class distribution). However, your idea sounds super interesting. I'm not aware of any research paper that applies such training objective, so it's use-cases still need to be verified :)

**Reply by rajivsam** [reply]
Is there a weighted cross entropy loss in this package?Any models that treat the sample weight as a parameter rather than a hyper-parameter.?

**Reply by rusty1s** [reply]
Is that even possible? I guess in this case the neural network just tries to minimize the weights to zero to obtain zero loss.

**Reply by rajivsam** [reply]
I don't know, and hence the question. I do remember adapting the weights in cost-sensitive learning ( a false positive may not be as bad as a false negative, but costs for these vary by example) was of interest in some problems - I came across this work:
https://proceedings.neurips.cc/paper/2019/file/e58cc5ca94270acaceed13bc82dfedf7-Paper.pdf
Cost sensitive classification has practical implications:
https://www.csd.uwo.ca/~xling/papers/cost_sensitive.pdf
Was curious to understand if this has made its way to the graph machine-learning world as well.  Thanks for the answer.
